kaizhilam/ProjectCreature
# Cooldowns

* checks in calCDTime to make sure when maxTime is 0, that slot's CD mask is gone and checks it doesn't try divide by 0 when currentTime is 0
* When slot stores item, it updates the CD timers
* Can now prompt slot to see if its on cooldown
* When swapping items around in hotbar, swaps their cooldown values too
* When swapping items out of hotbar, deactivates their cooldown too
* When item taken from inv and put in hotbar, will penalize player by running its cooldown (stops player being able to fast switch)
* Weapons picked up from the ground however, should be able to be used instantly
* Can now prompt InventoryManager to see if current weapon holding is on cooldown
* When player switches weapons, their ability function will change to the new weapons'
* prints to screen when player tries to right click with no weapon
* Players ability method only runs when its off cooldown

--Bugfixes--

* wielded player would briefly change from dash before going into walk wield

--How to test this branch--

* Pickup knife, press rmb. The cooldown should start (10s) and knife message should print to log
* Pickup armbow, press rmb. (5s cooldown) armbow ability message should print to log
* Switching armbow and knife when they're both on cooldown should swap their cooldown values
* Weapons on cooldown put into inv should lose their cooldown, and the slot should refresh
* New Weapons put into hotbar should play their cooldown immediately
* Player shouldn't be able to use ability when its on cooldown (no msg should print to log)
* Go into any weapons prefab and change their cooldown time on their script - it's new cooldown time should change to that value



BUG - boxcollider interfering with raycast for picking up items
